### PR TITLE
MAUI 874348 - UG documentation for Serialization in MAUI Image Editor

### PR DESCRIPTION
**Pull request by Shalini-Ashokan**
### Description
UG documentation for Serialization in MAUI Image Editor

### CODE REVIEW

File: MAUI/ImageEditor/serialization.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+title: Serialization in .NET MAUI Image Editor Control | Syncfusion
+description: Learn about serialization support in the Syncfusion .NET MAUI ImageEditor(SfImageEditor) control and more.
+platform: maui
+control: SfImageEditor
+documentation: ug
+---
+
+# Serialization in .NET MAUI Image Editor (SfImageEditor)
+
+The Image Editor control provides support to serialize and deserialize the annotations, free hand drawing, text and toolbar settings. You can save the current state of the image editor and load it back when it is needed.
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:12]: The Image Editor control provides support to serialize and deserialize the shape, text and pen annotations along with their settings. You can save the current state of the image editor annotations and load it back when it is needed.

File: MAUI/ImageEditor/serialization.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Serialization
+
+The `Serialize()` method is used to serialize the current edits of shapes. It allows you to store the SfImageEditor annotation settings by passing the stream as a parameter to the `Serialize` method.
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:16]: The `Serialize` method is used to serialize the current edits of shapes. It allows you to store the SfImageEditor annotations to the stream by passing the stream as a parameter to the `Serialize` method.

File: MAUI/ImageEditor/serialization.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        <Button Grid.Row="1"
+                Text="Serialize"
+                Clicked="OnSerializeClicked" />
+    </Grid>  
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+    
+    MemoryStream memoryStream;
+    private void OnSerializeClicked(object sender, EventArgs e)
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:33]: private void OnSerializeClicked(object sender, EventArgs e)
    {
        MemoryStream memoryStream = new MemoryStream();
        this.imageEditor.Serialize(memoryStream);
    }

File: MAUI/ImageEditor/serialization.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        this.memoryStream = new MemoryStream();
+        this.imageEditor.Serialize(memoryStream);
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Deserialization
+
+The Deserialize() method is used to deserialize the edits over an image. It allows you to reload the SfImageEditor control with the settings available in the stream.
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:44]: The `Deserialize` method is used to deserialize the annotations over an image. It allows you to reload the SfImageEditor control with the annotations available in the stream.

File: MAUI/ImageEditor/serialization.md (hunk 3 above)
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:33]: add example that creates a xml file in local storage and convert that into stream.

Use framework file picker concept

File: MAUI/ImageEditor/serialization.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                                   Source="image.jpeg" />
+        <Button Grid.Row="1"
+                Text="Deserialize"
+                Clicked="OnDeserializeClicked" />
+    </Grid>  
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+    private void OnDeserializeClicked(object sender, EventArgs e)
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:60]: add example which loads data from local file

File: maui-toc.html [hunk 6]
```diff
@@ -601,6 +601,7 @@
             <li><a href="/maui/ImageEditor/events">Events</a></li>
 		    <li><a href="/maui/ImageEditor/localization">Localization</a></li>
             <li><a href="/maui/ImageEditor/accessibility">Accessibility</a></li>
+            <li><a href="/maui/ImageEditor/serialization">Serialization</a></li>
```
Comment (karthikraja-arumugam) [inline maui-toc.html:604]: move topic before localization

**Review by karthikraja-arumugam — CHANGES_REQUESTED** [review]

File: MAUI/ImageEditor/serialization.md (hunk 1 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: Updated

File: MAUI/ImageEditor/serialization.md (hunk 4 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: Updated

File: MAUI/ImageEditor/serialization.md (hunk 3 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md:32]: Added example as you mentioned

File: MAUI/ImageEditor/serialization.md (hunk 5 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md:68]: Added example and loads data from local file

File: maui-toc.html (hunk 6 above)
Comment (Shalini-Ashokan) [reply maui-toc.html]: Moved

File: MAUI/ImageEditor/serialization.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+    
+    private void OnSerializeClicked(object sender, EventArgs e)
+    {
+        string filePath = Path.Combine(FileSystem.Current.CacheDirectory, "ImageEditor.xml");
+        if (!File.Exists(filePath))
+        {
+            using (FileStream fs = File.Create(filePath))
+            {
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:38]: Code is not correct

File: MAUI/ImageEditor/serialization.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    private void OnDeserializeClicked(object sender, EventArgs e)
+    {
+        var result = await FilePicker.PickAsync(new PickOptions
+        {
+            PickerTitle = "Select an XML file"
+        });
+        
+        if (result != null)
+        {
+            MemoryStream memoryStream = new MemoryStream();
```
Comment (karthikraja-arumugam) [inline MAUI/ImageEditor/serialization.md:77]: use can directly use stream object no need to copy it to MemoryStream

File: MAUI/ImageEditor/serialization.md (hunk 7 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: Changed a code example

File: MAUI/ImageEditor/serialization.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+documentation: ug
+---
+
+# Serialization in .NET MAUI Image Editor (SfImageEditor)
+
+The Image Editor control provides support to serialize and deserialize the shape, text and pen annotations along with their settings. You can save the current state of the image editor annotations and load it back when it is needed.
+
+## Serialization
+
+The `Serialize` method is used to serialize the current edits of shapes. It allows you to store the `SfImageEditor` annotations to the stream by passing the stream as a parameter to the `Serialize` method.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: Updated

File: MAUI/ImageEditor/serialization.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                Clicked="OnDeserializeClicked" />
+    </Grid>  
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+    private void OnDeserializeClicked(object sender, EventArgs e)
+    {
+        var result = await FilePicker.PickAsync(new PickOptions
+        {
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: removed title code

File: MAUI/ImageEditor/serialization.md (hunk 8 above)
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md]: Used directly stream

File: MAUI/ImageEditor/serialization.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+            using (Stream stream = await result.OpenReadAsync())
+            {
+                await stream.CopyToAsync(memoryStream);
+            }
+        
+            this.imageEditor.Deserialize(memoryStream);
+        }
+    }
+
+{% endhighlight %}
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/ImageEditor/serialization.md:75]: Added

**Pull request closed without merging**